sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -488,9 +489,24 @@ public void updateIedNameBasedOnLnode(SCL scl) {
                                                                 && Objects.equals(tlNode.getLnInst(), tCompasFlow.getExtReflnInst())
                                                                 && Utils.lnClassEquals(tlNode.getLnClass(), tCompasFlow.getExtReflnClass())
                                                                 && Objects.equals(tlNode.getPrefix(), tCompasFlow.getExtRefprefix()))
+                                                        .filter(tlNode -> {
+                                                            TCompasICDHeader tCompasICDHeader = PrivateUtils.extractCompasPrivate(tlNode, TCompasICDHeader.class)
+                                                                    .orElseThrow(() -> new ScdException(("The substation LNode with following attributes : IedName:%s / LdInst:%s / LnClass:%s / LnInst:%s  " +
+                                                                            "does not contain the needed private (COMPAS - ICDHeader)")
+                                                                            .formatted(tlNode.getIedName(), tlNode.getLdInst(), tlNode.getLnClass().getFirst(), tlNode.getLnInst())));
+                                                            return Objects.equals(tCompasFlow.getFlowSourceIEDType(), tCompasICDHeader.getIEDType())
```

Comment:
It states that it shouldn't be used as a criteria if there are no properties.
```suggestion
                                                            return tCompasFlow.isSetFlowSourceIEDType() && Objects.equals(tCompasFlow.getFlowSourceIEDType(), tCompasICDHeader.getIEDType())
```

Reply:
Indeed, issue has to be changed according to last owner POV which states to consider them anytime. @hkaroun